core/src/main/java/io/aiven/klaw/validation/TopicRequestValidatorImpl.java

```diff
@@ -1,21 +1,244 @@
+      if (promotionOrderCheck) {
+        updateConstraint(
+            constraintValidatorContext,
+            "Failure. Please request for a topic first in "
+                + topicControllerService.getEnvDetails(syncCluster).getName()
+                + " cluster.");
+        return false;
+      }
+    }
+
     return true;
-    // TODO move logic from service layer to here for validating topic request
+  }
+
+  private boolean validateTopicConfigParameters(
+      TopicRequestModel topicRequestReq, ConstraintValidatorContext constraintValidatorContext) {
+    log.debug("Into validateTopicConfigParameters");
+
+    String topicPrefix = null, topicSuffix = null;
+    String otherParams =
+        topicControllerService.getEnvDetails(topicRequestReq.getEnvironment()).getOtherParams();
+    String[] params;
+    try {
+      if (otherParams != null) {
+        params = otherParams.split(",");
+        for (String param : params) {
+          if (param.startsWith("topic.prefix")) {
+            topicPrefix = param.substring(param.indexOf("=") + 1);
+          } else if (param.startsWith("topic.suffix")) {
+            topicSuffix = param.substring(param.indexOf("=") + 1);
+          }
+        }
+      }
+    } catch (Exception e) {
+      log.error("Unable to set topic partitions, setting default from properties.", e);
+    }
+
+    try {
+      if (topicPrefix != null
+          && topicPrefix.length() > 0
+          && !topicRequestReq.getTopicname().startsWith(topicPrefix)) {
+        log.error(
+            "Topic prefix {} does not match. {}", topicPrefix, topicRequestReq.getTopicname());
+        updateConstraint(
+            constraintValidatorContext,
+            "Topic prefix does not match. " + topicRequestReq.getTopicname());
+        return false;
+      }
+
+      if (topicSuffix != null
+          && topicSuffix.length() > 0
+          && !topicRequestReq.getTopicname().endsWith(topicSuffix)) {
+        log.error(
+            "Topic suffix {} does not match. {}", topicSuffix, topicRequestReq.getTopicname());
+        updateConstraint(
+            constraintValidatorContext,
+            "Topic suffix does not match. " + topicRequestReq.getTopicname());
+        return false;
+      }
+    } catch (Exception e) {
+      log.error("Unable to set topic partitions, setting default from properties.", e);
+      updateConstraint(
+          constraintValidatorContext, "Cluster default parameters config missing/incorrect.");
+      return false;
+    }
+
+    return true;
+  }
+
+  private void updateConstraint(
+      ConstraintValidatorContext constraintValidatorContext, String errorMessage) {
+    constraintValidatorContext
+        .buildConstraintViolationWithTemplate(errorMessage)
+        .addConstraintViolation()
+        .disableDefaultConstraintViolation();
+  }
+
+  private boolean checkInPromotionOrder(String envId, String orderOfEnvs) {
+    List<String> orderedEnv = Arrays.asList(orderOfEnvs.split(","));
```

Comment:
same question about env
in case of system env it's ok to have envvar containing `,` however in that case it will be broken here

Reply:
there is a null check before this method invocation. that should be ok I guess.
`if (null != orderOfEnvs)`